File: native-ios/JuanbertosPOS/Models/Payment.swift
import Foundation

struct PaymentIntent: Codable, Sendable {
    let client_secret: String
    let payment_intent_id: String
    let amount: Double
}

struct PaymentStatus: Codable, Sendable {
    let status: PaymentStatusType
    let payment_intent_id: String?
    let amount: Double?
}

struct CreatePaymentIntentRequest: Codable, Sendable {
    let order_id: Int
    let tip: Double?
}

struct ConfirmPaymentRequest: Codable, Sendable {
    let order_id: Int
    let payment_intent_id: String
}

struct CashPaymentRequest: Codable, Sendable {
    let order_id: Int
    let tip: Double?
    let amount_received: Double?
}

struct RefundPaymentRequest: Codable, Sendable {
    let order_id: Int
    let amount: Double?
}

struct SplitPaymentSplit: Codable, Sendable {
    let payment_method: PaymentMethod
    let amount: Double
    let tip: Double?
    let item_ids: [Int]?
}

struct SplitPaymentRequest: Codable, Sendable {
    let order_id: Int
    let split_type: String // "even" | "by_item" | "by_amount"
    let splits: [SplitPaymentSplit]
}

struct OrderPayment: Codable, Identifiable, Sendable {
    let id: Int
    var order_id: Int
    var payment_method: PaymentMethod
    var amount: Double
    var tip: Double
    var payment_intent_id: String?
    var status: String
}

File: apps/pos/native-android/app/src/main/java/com/desktopkitchen/pos/models/Payment.kt
package com.desktopkitchen.pos.models

import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class PaymentIntent(
    val client_secret: String,
    val payment_intent_id: String,
    val amount: Double
)

@JsonClass(generateAdapter = true)
data class PaymentStatus(
    val status: String,
    val payment_intent_id: String? = null,
    val amount: Double? = null
)

@JsonClass(generateAdapter = true)
data class CreatePaymentIntentRequest(
    val order_id: Int,
    val tip: Double? = null
)

@JsonClass(generateAdapter = true)
data class ConfirmPaymentRequest(
    val order_id: Int,
    val payment_intent_id: String
)

@JsonClass(generateAdapter = true)
data class CashPaymentRequest(
    val order_id: Int,
    val tip: Double? = null,
    val amount_received: Double? = null
)

@JsonClass(generateAdapter = true)
data class PaymentSplit(
    val payment_method: String,
    val amount: Double,
    val tip: Double = 0.0
)

@JsonClass(generateAdapter = true)
data class SplitPaymentRequest(
    val order_id: Int,
    val split_type: String = "mixed",
    val splits: List<PaymentSplit>
)

@JsonClass(generateAdapter = true)
data class SplitPaymentResponse(
    val success: Boolean,
    val message: String? = null,
    val splits_count: Int? = null
)
